internal/database/conf.go

```diff
@@ -141,15 +141,25 @@ SELECT
 	id,
 	author_user_id,
 	contents,
-    redacted_contents,
+	redacted_contents,
 	created_at,
 	updated_at
-FROM critical_and_site_config
-WHERE (%s)
+FROM (
+	SELECT
+		*,
+		lag(redacted_contents) OVER (ORDER BY id) AS prev_redacted_contents
```

Comment:
Nitpick, but I do like SQL keywords/functions being upper-case (`SELECT`, `COUNT`, `FIlTER`, ...)

```suggestion
		LAG(redacted_contents) OVER (ORDER BY id) AS prev_redacted_contents
```